flow/project.py

```diff
@@ -1491,11 +1491,17 @@ def run(self, jobs=None, names=None, pretend=False, np=None, timeout=None, num=N
             Show a progress bar during execution.
         :type progess:
             bool
-        :param shuffle:
-            Shuffle operations prior to each execution pass.
-        :type shuffle:
-            bool
+        :param order:
+            Specify the order of operations, possible values are:
+                * None or 'none' (no reordering, do not assume any order)
+                * 'by-job'  (operations are grouped by job)
+                * 'round-robin' (order operations in round-robin fashion by job)
+                * 'random'    (shuffle the execution order randomly)
+                * callable specify a key-callable by which to order operations
+            The default value is None.
         """
+        from itertools import groupby
```

Comment:
I am not sure why the imports are going here. If the purpose is conditional importing, shouldn't it be further down in the if statements? Otherwise, I think it should go to the top with the other imports as is typical python form.